src/lib/Metadata/Finalizers/BaseMembersFinalizer.cpp

```diff
@@ -116,7 +122,7 @@ inheritBaseMembers(
 
         // Check if derived class has a member that shadows the base member
         auto shadowIt = std::ranges::find_if(
```

Comment:
Oh... Here's the relevant point. Yes. We can just call "allMembers" here to get all members of derived. This should be a lazy operation so there's no extra non-constant cost in just calling it again.

The only unnecessary extra-cost we have here is the cost of iterating all kinds of members while we want the functions in different access modes. We can fix that by just creating a lazy range that only contains the functions, or by iterating the functions for each access mode.